### ISSUE TITLE
Sampler creation on one 12 second file takes 5-8seconds on iphone 8 plus.

### ISSUE THREAD

**Issue by quantamrhino** [AudioKit/AudioKit#2286]
I'm on v5 develop. 

I have 1 AVAudioFile of about 5 seconds in length.
The code to load the file from bundle to avaudiofile takes 0.0007 seconds.

            let fullpath = Bundle.main.path(forResource: path, ofType: "")
            let file = try AVAudioFile(forReading: URL(fileURLWithPath: fullpath!))

Then, I call Sampler(descriptor, file).
For a 4 second file, Sampler(descriptor, file) takes 0.63 to 0.8 seconds.
for a 12 second file, Sampler(descriptor, file) takes 5.1 to 8.3 seconds 
This is on a physical device - iphone 8 plus, on ios14.

    func initDescriptor(note: MIDINoteNumber) -> SampleDescriptor {
        return SampleDescriptor(noteNumber: Int32(note),
                              noteFrequency: Float(PolyphonicNode.tuningTable.frequency(forNoteNumber: note)),
                              minimumNoteNumber: -1,
                              maximumNoteNumber: -1,
                              minimumVelocity: -1,
                              maximumVelocity: -1,
                              isLooping: false,
                              loopStartPoint: 0,
                              loopEndPoint: 1,
                              startPoint: 0,
                              endPoint: 0)
    }

        let a = Date()
        let file = Audio.loadAudioFile("12secondsample.wav")!
        let b = Date()
        sampler = Sampler(sampleDescriptor: initDescriptor(note: 60), file: file)
        print("create sampler time : \(Date().timeIntervalSince(b)), file duration: \(file.duration), file load: \(b.timeIntervalSince(a))")

For a file of duration 0.1 seconds, the load and creation happen in < 0.001 seconds. 
The only part that scales with the size of the sample is the loadAudioFile.
This worked very fast in v4. (the loadAudioFiles() part -- i had one sampler that'd unload and reload samples on the fly when the instrument setting changed and things worked very well).

Please help Aure sir!

**Comment by quantamrhino** [comment]
I tried with 12 files of 3 seconds each. Sampler creation took 9 seconds.
These messages are spaced in that interval:
AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39:Created buffer with format <AVAudioFormat 0x6000038437a0:  2 ch,  48000 Hz, Float32, non-inter> (AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39

**Comment by quantamrhino** [comment]
For the same files, AppleSampler takes about 0.02 seconds to load.

**Comment by aure** [comment]
Please provide a project that demonstrates this, really output thing load times as its output.

**Comment by quantamrhino** [comment]
[SamplerComparison.zip](https://github.com/AudioKit/AudioKit/files/5287744/SamplerComparison.zip)

5 Files of nearly 20 second samples, 48KHz.
Here's a project. Apple Sampler create/load time: 0.2 seconds.
AudioKit sampler create/load time: 12.35 seconds! ~62 times slower...
(On device - 7 plus and simulator - iphone SE 1 simulator, takes roughly the same time).

Here's the output log of the code after you press "Run Code" button.

avfile load time: 0.0016061067581176758
avfile load time: 0.0009270906448364258
avfile load time: 0.0008840560913085938
avfile load time: 0.0007960796356201172
avfile load time: 0.000744938850402832
2020-09-27 10:21:30.938579+0530 SamplerComparison[33761:3353816] [general] AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39:Created buffer with format <AVAudioFormat 0x600000c503c0:  2 ch,  48000 Hz, Float32, non-inter> (AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39)
2020-09-27 10:21:33.365691+0530 SamplerComparison[33761:3353816] [general] AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39:Created buffer with format <AVAudioFormat 0x600000c5d9f0:  2 ch,  48000 Hz, Float32, non-inter> (AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39)
2020-09-27 10:21:35.729377+0530 SamplerComparison[33761:3353816] [general] AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39:Created buffer with format <AVAudioFormat 0x600000c5d3b0:  2 ch,  48000 Hz, Float32, non-inter> (AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39)
2020-09-27 10:21:38.090022+0530 SamplerComparison[33761:3353816] [general] AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39:Created buffer with format <AVAudioFormat 0x600000c5a8a0:  2 ch,  48000 Hz, Float32, non-inter> (AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39)
2020-09-27 10:21:40.636363+0530 SamplerComparison[33761:3353816] [general] AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39:Created buffer with format <AVAudioFormat 0x600000c5db30:  2 ch,  48000 Hz, Float32, non-inter> (AVAudioFile+Utilities.swift:toAVAudioPCMBuffer():39)

**av load time : 0.005423903465270996, apple sampler create time : 0.20706307888031006, akSampler create time: 12.349593997001648**

**Comment by aure** [comment]
@getdunne Maybe we could do a sleuth session sometime on my computer?

**Comment by aure** [comment]
@quantamrhino We'll take a look but I encourage you try to figure out the problem yourself. Use Apple's "Instruments" app to get more detail on where that time is being spent.

**Comment by quantamrhino** [comment]
I'll try and learn about instruments. I'm taking an intro to swift programming course, but this is covered in the next semester i think. To be honest, i need to demo my code to get a passing grade :-( I'm trying with apple sampler for now because the deadline is in 3 weeks. 
I'll try and learn about instruments etc as soon as the final project is done. then we have a 2 week break in mid november for diwali.
Thank you for all your help and for audiokit - i hope my end project wows the prof :)

**Comment by aure** [comment]
So, the author of the AudioKit Sampler, @getdunne and I took a look at this this afternoon. When this was first written, it relied on AKAudioFile type to get the floatChannelData. Since, then we moved onto AVAudioFile. This work, but it seems like a lot of time is spent grabbing the floatChannelData out in Swift and then processing it for the Sampler's loadAudioFile function. Not sure what can be done about this.
